libraries/Microsoft.Bot.StreamingExtensions/Payloads/Assemblers/ContentStreamAssembler.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Bot.StreamingExtensions.Payloads
+{
+    internal class ContentStreamAssembler : PayloadAssembler
+    {
+        private readonly IStreamManager _streamManager;
+
+        public ContentStreamAssembler(IStreamManager streamManager, Guid id)
+            : base(id)
+        {
+            ContentType = null;
+            ContentLength = null;
+            _streamManager = streamManager;
+        }
+
+        public ContentStreamAssembler(IStreamManager streamManager, Guid id, string type, int? length)
+            : this(streamManager, id)
+        {
+            ContentType = type;
+            ContentLength = length;
```

Comment:
null check everything. Get rid of the null programming. (nullable lenght? That doesn't make any logical sense)

Reply:
The nullable length is to support infinite streams, literally streams with no known length.